cli/src/cli.ts

```diff
@@ -1,41 +1,45 @@
 import { Range } from 'immutable'
+import type { Server } from 'node:http'
 
 import type { TrainerLog, data, Task } from '@epfml/discojs-core'
-import { Disco, TrainingSchemes } from '@epfml/discojs-core'
+import { Disco, TrainingSchemes, aggregator as aggregators, client as clients } from '@epfml/discojs-core'
+import { getClient, startServer } from '@epfml/disco-server'
 
-import { startServer, saveLog } from './utils'
+import { saveLog } from './utils'
 import { getTaskData } from './data'
 import { args } from './args'
 
 const NUMBER_OF_USERS = args.numberOfUsers
 const TASK = args.task
 
-const infoText = `\nStarted federated training of ${TASK.taskID}`
+const infoText = `\nStarted federated training of ${TASK.id}`
 console.log(infoText)
 
 console.log({ args })
 
-async function runUser (task: Task, url: URL, data: data.DataSplit): Promise<TrainerLog> {
+async function runUser (task: Task, server: Server, data: data.DataSplit): Promise<TrainerLog> {
+  const client = await getClient(clients.federated.FederatedClient, server, task, new aggregators.MeanAggregator(TASK))
```

Comment:
What's the point of passing a whole client to Disco rather than the url?

Reply:
getting the URL out of a server is non-trivial, as you can see from `getClient` impl. but now that you mention it, passing the server in full here disallow connecting to an existing server. I'll update it so that `startServer` actually returns the URL to connect to and get rid of `getClient`.